source/conf.ts

```diff
@@ -0,0 +1,531 @@
+			}
+
+			this._migrate(options.migrations, options.projectVersion);
+		}
+	}
+
+	/**
+	Get an item.
+
+	@param key - The key of the item to get.
+	@param defaultValue - The default value if the item does not exist.
+	*/
+	get<TKey extends keyof T>(key: TKey | string, defaultValue?: T[TKey] | any): T[TKey] | undefined {
+		if (this._options.accessPropertiesByDotNotation) {
+			return this._get(key, defaultValue);
+		}
+
+		return key in this.store ? this.store[key] : defaultValue;
+	}
+
+	set<TKey extends keyof T>(key: Partial<T> | TKey | string, value?: T[TKey] | any): void {
+		if (typeof key !== 'string' && typeof key !== 'object') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\` or \`object\`, got ${typeof key}`);
+		}
+
+		if (typeof key !== 'object' && value === undefined) {
+			throw new TypeError('Use `delete()` to clear values');
+		}
+
+		if (this._containsReservedKey(key)) {
+			throw new TypeError(`Please don't use the ${INTERNAL_KEY} key, as it's used to manage this module internal operations.`);
+		}
+
+		const {store} = this;
+
+		const set = (key: string, value?: T[TKey] | T): void => {
+			checkValueType(key, value);
+			if (this._options.accessPropertiesByDotNotation) {
+				dotProp.set(store, key, value);
+			} else {
+				store[key] = value as T;
+			}
+		};
+
+		if (typeof key === 'object') {
+			const object = key;
+			for (const [key, value] of Object.entries(object)) {
+				set(key, value);
+			}
+		} else {
+			set(key, value);
+		}
+
+		this.store = store;
+	}
+
+	/**
+	Check if an item exists.
+
+	@param key - The key of the item to check.
+	*/
+	has<Tkey extends keyof T>(key: Tkey | string): boolean {
+		if (this._options.accessPropertiesByDotNotation) {
+			return dotProp.has(this.store, key as string);
+		}
+
+		return key in this.store;
+	}
+
+	/**
+	Reset items to their default values, as defined by the `defaults` or `schema` option.
+
+	@param keys - The keys of the items to reset.
+	*/
+	reset<TKey extends keyof T>(...keys: Array<TKey | string>): void {
+		for (const key of keys) {
+			if (this._defaultValues[key]) {
+				this.set(key, this._defaultValues[key]);
+			}
+		}
+	}
+
+	/**
+	Delete an item.
+
+	@param key - The key of the item to delete.
+	*/
+	delete<TKey extends keyof T>(key: TKey): void {
+		const {store} = this;
+		if (this._options.accessPropertiesByDotNotation) {
+			dotProp.delete(store, key as string);
+		} else {
+			delete store[key];
+		}
+
+		this.store = store;
+	}
+
+	/**
+	Delete all items.
+	*/
+	clear(): void {
+		this.store = createPlainObject();
+	}
+
+	/**
+	Watches the given `key`, calling `callback` on any changes.
+
+	@param key - The key wo watch.
+	@param callback - A callback function that is called on any changes. When a `key` is first set `oldValue` will be `undefined`, and when a key is deleted `newValue` will be `undefined`.
+	*/
+	onDidChange<TKey extends keyof T>(
+		key: TKey,
+		callback: OnDidChangeCallback<T[TKey]>
+	): () => void {
```

Comment:
What do you think about making the return function here a named type so it's clearer what it does?

Reply:
Yes, that would be cleaner 👍